Packages/com.unity.inputsystem/InputSystem/Events/IMECompositionEvent.cs

```diff
@@ -132,5 +95,62 @@ object IEnumerator.Current
                 get { return Current; }
             }
         }
+
+        public int Count
+        {
+            get
+            {
+                return size;
+            }
+        }
+
+        public unsafe char this[int index]
+        {
+            get
+            {
+                if (index >= Count || index < 0)
+                    throw new IndexOutOfRangeException();
+
+                fixed(char* ptr = buffer)
+                {
+                    return *(ptr + index);
+                }
+            }
+        }
+
+        [FieldOffset(0)]
+        int size;
+
+        [FieldOffset(sizeof(int))]
+        fixed char buffer[LowLevel.IMECompositionEvent.kIMECharBufferSize];
+
+        public IMECompositionString(string characters)
+        {
+            Debug.Assert(characters.Length < LowLevel.IMECompositionEvent.kIMECharBufferSize);
```

Comment:
If it's a public API, it has to throw. `Debug.Asserts` are only okay for when we want to guard against our own code messing up.